### ISSUE TITLE
(!) Plugin svelte: A11y: '#' is not a valid href attribute

### ISSUE THREAD

**Issue by chovy** [ItalyPaleAle/svelte-spa-router#66]
(!) Plugin svelte: A11y: '#' is not a valid href attribute

how should I handle this? I have a link with an onclick handler

**Comment by ItalyPaleAle** [comment]
This is not an issue with the router. It’s Svelte telling you that links to # are bad for accessibility.

If you want to link to the home page, use #/ for href. If you want a link that does nothing, use “javascript:void(0)” for href. Or just don’t use an anchor tag and use any other kind of tag :)

**Comment by chovy** [comment]
without href attribute you don't get the proper cursor.

**Comment by emiola** [comment]
You can also prevent the warning with a comment:

```
    <!-- svelte-ignore a11y-missing-attribute -->
    <a>...</a>
```

**Comment by ItalyPaleAle** [comment]
Cursor can be changed with CSS cursor rule

**Comment by IDDT** [comment]
`<a href={'#'}></a>` also removes the warning.

**Comment by CherryDT** [comment]
> If you want a link that does nothing, use “javascript:void(0)” for href.

The problem with this is that with strict content security policy (which is recommended) it will generate CSP warnings and send CSP audit reports for no reason.

**Comment by shijiezhou1** [comment]
> `<a href={'#'}></a>` also removes the warning.

This is awesome I found it super helpful

**Comment by KodiakMack** [comment]
> `<a href={'#'}></a>` also removes the warning.

Worked awesome !

**Comment by mholtzhausen** [comment]
In most cases where this is used, it is intercepted with `on:click` handler to perform some action.
There is an accessibility issue with providing a link, but a screen-reader can't read where it is going - so links must always be clear in what they're attempting to do.

When you intercept an on:click action like that though, you're most likely not trying to really create a link, you're just using it because it is a convenient click-able to trigger your action. 

The real solution here is to use a button. https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role

**Comment by b1ek** [comment]
> use “javascript:void(0)”

don't. that would break most content security policies
